ale_linters/go/gobuild.vim

```diff
@@ -39,15 +40,21 @@ function! ale_linters#go#gobuild#GetMatches(lines) abort
 endfunction
 
 function! ale_linters#go#gobuild#Handler(buffer, lines) abort
+    let l:dir = expand('#' . a:buffer . ':p:h')
     let l:output = []
 
     for l:match in ale_linters#go#gobuild#GetMatches(a:lines)
-        " Omit errors from imported go packages
-        if !ale#path#IsBufferPath(a:buffer, l:match[1])
+        "
+        " Get the list of parent paths from the current filename
+        let l:paths = ale#path#Upwards(expand('#' . a:buffer . ':p'))
+        " Only see errors from this package (directory), ignore errors from imported packages
+        if !l:dir is# l:paths[1]
```

Comment:
I'm confused. Above you're getting the directory for the filename. Then here you're getting the directory for the same filename again, but in a different and more expensive way, for every iteration?

Reply:
It should have been outside of the loop...didn't catch that

As for why I was getting the directory again...I wanted to compare the error path against the current buffer path to only get things in the same directory. The error path should have been coming from `l:match` I guess but the whole line of thought becomes irrelevant due to the suggestion above..